### PR TITLE
[Runtime] Move some of the changes back to int.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Some of the fixes done for the warnings have breaking changes. Move back
to int and ensure that we do not have any compilation errors (we are
using -Werror).

Fixes: https://github.com/xamarin/xamarin-macios/issues/7509

### CODE REVIEW

File: runtime/trampolines-invoke.m [hunk 1]
```diff
@@ -598,17 +598,17 @@
...
 					LOGZ (" writing back managed array %p to argument at index %i (%p)\n", value, i + 1, arg);
 				} else {
-					exception_gchandle = xamarin_get_exception_for_parameter (8030, 0, "Unable to marshal the out/ref parameter", sel, method, p, i, false);
+					exception_gchandle = xamarin_get_exception_for_parameter (8030, 0, "Unable to marshal the out/ref parameter", sel, method, p, (int) i, false);
 					goto exception_handling;
 				}
 				*(NSObject **) arg = obj;
 			} else {
-				exception_gchandle = xamarin_get_exception_for_parameter (8030, 0, "Unable to marshal the out/ref parameter", sel, method, p, i, false);
+				exception_gchandle = xamarin_get_exception_for_parameter (8030, 0, "Unable to marshal the out/ref parameter", sel, method, p, (int) i, false);
```
Comment (spouliot) [inline runtime/trampolines-invoke.m]: would not this be simpler (less changes) if `int i` was used ?

Comment (mandel-macaque) [reply runtime/trampolines-invoke.m]: Surprisingly, no.

**Comment by rolfbjarne** [comment]
https://github.com/xamarin/maccore/issues/2072 is still happening, so this is not complete. My first guess would be that it's the changes in trampolines-arm64.m that is causing https://github.com/xamarin/maccore/issues/2072.

**Comment by mandel-macaque** [comment]
@rolfbjarne these results do look more promising.

**Comment by rolfbjarne** [comment]
@mandel-macaque so that didn't fix https://github.com/xamarin/maccore/issues/2072 😒

**Comment by spouliot** [comment]
@monojenkins backport to d16-5

**Pull request merged**